modules/javafx.graphics/src/main/java/javafx/animation/Interpolator.java

```diff
@@ -256,6 +266,66 @@ public static Interpolator TANGENT(Duration t, double v) {
         return new NumberTangentInterpolator(t, v);
     }
 
+    /**
+     * Specifies the step position of a step interpolator.
+     * <p>
+     * The step position determines the location of rise points in the input progress interval, which are the
+     * locations on the input progress axis where the output progress value jumps from one step to the next.
+     *
+     * @since 22
+     */
+    public enum StepPosition {
+        /**
+         * The interval starts with a rise point when the input progress value is 0.
+         */
+        START,
+
+        /**
+         * The interval ends with a rise point when the input progress value is 1.
+         */
+        END,
+
+        /**
+         * All rise points are within the open interval (0..1).
+         */
+        BOTH,
+
+        /**
+         * The interval starts with a rise point when the input progress value is 0,
+         * and ends with a rise point when the input progress value is 1.
+         */
+        NONE
+    }
+
+    /**
+     * Built-in interpolator instance that is equivalent to {@code STEPS(1, StepPosition.START)}.
+     *
+     * @since 22
+     */
+    public static Interpolator STEP_START = STEPS(1, StepPosition.START);
+
+    /**
+     * Built-in interpolator instance that is equivalent to {@code STEPS(1, StepPosition.END)}.
+     *
+     * @since 22
+     */
+    public static Interpolator STEP_END = STEPS(1, StepPosition.END);
+
+    /**
+     * Creates a step interpolator that divides the input time into a series of intervals, each
+     * interval being equal in length, where each interval maps to a constant output time value.
+     * The output time value is determined by the {@link StepPosition}.
+     *
+     * @param intervals the number of intervals in the step interpolator
+     * @param position the {@code StepPosition} of the step interpolator
+     * @return a new step interpolator
+     *
+     * @since 22
+     */
+    public static Interpolator STEPS(int intervals, StepPosition position) {
```

Comment:
This doesn't specify what happens when `position` is `null`, and is also missing the restrictions on `intervals`.